### ISSUE TITLE
Shell files not found or linted when VALIDATE_SHELL_SHFMT enabled

### ISSUE THREAD

**Issue by davidrimprobable** [github/super-linter#2068]
## Describe the bug

When explicitly setting `VALIDATE_SHELL_SHFMT: true` and `VALIDATE_ALL_CODEBASE: true` no shell files are automatically found or linted despite there being `*.sh` files present in the repository 

## Expected behavior
Expected *.sh files to be found and auto linted

## Steps to Reproduce
Have `.sh` files in directories nested under $PWD (e.g. I have `./workflow/deploy.sh`)

Run with 

```
docker run \
  -e RUN_LOCAL=true \
  -e DISABLE_ERRORS=false \
  -e ERROR_ON_MISSING_EXEC_BIT=true \
  -e LINTER_RULES_PATH=. \
  -e MULTI_STATUS=false \
  -e VALIDATE_ALL_CODEBASE=true \
  -e VALIDATE_SHELL_SHFMT=true \
  -v $(pwd):/tmp/lint \
 github/super-linter:slim-v4
```

there is no visible attempt to find or lint shell files

```
2021-10-25 09:57:52 [INFO]   --------------------------------------------
2021-10-25 09:57:52 [INFO]   Gathering GitHub information...
2021-10-25 09:57:52 [INFO]   NOTE: ENV VAR [RUN_LOCAL] has been set to:[true]
2021-10-25 09:57:52 [INFO]   bypassing GitHub Actions variables...
2021-10-25 09:57:52 [INFO]   Linting all files in mapped directory:[/tmp/lint]
2021-10-25 09:57:52 [INFO]   Successfully found:[GITHUB_TOKEN]
2021-10-25 09:57:52 [INFO]   --------------------------------------------
2021-10-25 09:57:52 [INFO]   Gathering user validation information...
2021-10-25 09:57:55 [INFO]   ----------------------------------------------
2021-10-25 09:57:55 [INFO]   User provided file:[/tmp/lint/.yaml-lint.yml] exists, setting rules file...
2021-10-25 09:57:59 [INFO]   ---------------------------------
2021-10-25 09:57:59 [INFO]   ------ File list to check: ------
2021-10-25 09:57:59 [INFO]   ---------------------------------
2021-10-25 09:58:30 [INFO]   ----------------------------------------------
2021-10-25 09:58:30 [INFO]   Successfully gathered list of files...
2021-10-25 09:58:31 [INFO]   ----------------------------------------------
2021-10-25 09:58:31 [INFO]   ----------------------------------------------
2021-10-25 09:58:31 [INFO]   The script has completed
2021-10-25 09:58:31 [INFO]   ----------------------------------------------
2021-10-25 09:58:31 [INFO]   ----------------------------------------------
2021-10-25 09:58:31 [NOTICE]   All file(s) linted successfully with no errors detected
2021-10-25 09:58:31 [INFO]   ----------------------------------------------
```

I've also attempted this after creating `my-shell-script.sh` at the repo root with

```shell
#!/bin/bash
${array[spaced string]}
```

as content, and also having marked it as executable - the result is the same.

Contrast this with `VALIDATE_JSON` for example, where 'linting [JSON] files' will be shown and lint will occur regardless of directory/nesting

## bash vs. shfmt?
If I run with `VALIDATE_BASH` shellcheck runs correctly in the same way I would have expected from shfmt

Is there a documentation hole here that shfmt is not _expected_ to trigger/run on its own in validate all codebase mode? Is there some kind of dependency between the bash validator and the shfmt one for searching for files?

**Comment by ferrarimarco** [comment]
Hi @davidrimprobable , can you run it locally again adding `-e ACTIONS_RUNNER_DEBUG=true \` and post the relevant output?

**Comment by davidrimprobable** [comment]
> Hi @davidrimprobable , can you run it locally again adding `-e ACTIONS_RUNNER_DEBUG=true \` and post the relevant output?

Sure: I made a test project with just `my-shell-script.sh` and ran 

- with `VALIDATE_SHELL_SHFMT` (did not work)
- with `VALIDATE_BASH` (works as expected)

both with `ACTIONS_RUNNER_DEBUG` enabled

Attached both logs for comparison:

[shfmt-notworking.log](https://github.com/github/super-linter/files/7581979/shfmt-notworking.log)

[bash-working.log](https://github.com/github/super-linter/files/7581981/bash-working.log)

**Comment by ferrarimarco** [comment]
I think the key is (from `shfmt-notworking.log`):

```
2021-11-22 14:39:34 [DEBUG]  No .editorconfig found at: /tmp/lint/.editorconfig. Skipping SHELL_SHFMT linting...
```

Reason is here:

https://github.com/github/super-linter/blob/fb8212608290c165785a9f4d6e90bad904d93690/lib/linter.sh#L992-L998

Rationale: https://github.com/mvdan/sh/blob/master/cmd/shfmt/shfmt.1.scd#description

TLDR: You need an `.editorconfig` file (even if empty) in the `LINTER_RULES_PATH` directory, which you set to `.` (= the root of your repository. See [shfmt docs](https://github.com/mvdan/sh/blob/master/cmd/shfmt/shfmt.1.scd#examples) for details about how to configure shfmt using an EditorConfig file.

Feel free to reopen if this doesn't solve your issue. :)

/cc @admiralAwkbar

**Comment by davidrimprobable** [comment]
This solves my issue but I think it is still a bug in that the docs should be updated or this behaviour changed - a config isn't mandatory for shfmt especially for detecting the errors in my test project.

Maybe you'd never want to run it without a config on a big project but I think the docs for `VALIDATE_SHELL_SHFMT` should call this out because it's not obvious

**Comment by ferrarimarco** [comment]
@davidrimprobable fair enough :)

Feel free to open a PR to add the missing guidance to the docs. Thanks!
